web/src/lib/utils/file-uploader.ts

```diff
@@ -127,3 +133,7 @@ async function fileUploader(asset: File, albumId: string | undefined = undefined
       return undefined;
     });
 }
+
+function isExternalUser() {
+  return window.location.href.includes('/share/');
```

Comment:
I don't think this is the correct way to identify public user, because you can be logged in as the owner of the shared link/album and view the `share` page content as well